src/quo2/components/network_amount.cljs

```diff
@@ -0,0 +1,85 @@
+  {:icon {:small 12 :medium 16 :large 20}
+   :container {:small 32 :medium 32 :large 48}
+   :label {:small :label :medium :paragraph-2 :large :paragraph-1}
+   :amount-text {:small :paragraph-2 :medium :paragraph-1 :large :heading-2}
+   :divider {:small {:width 1 :height 8}
+             :medium {:width 1 :height 10}
+             :large {:width 2 :height 12}}})
+
+(defn- format-eth-value [eth-value]
+  (str eth-value \space (i18n/label :t/eth)))
+
+(defn- amount-text [size eth-value]
+  (let [opts {:weight :medium
+              :size (get-in sizes [:amount-text size])
+              :style {:color (colors/theme-colors colors/black colors/white)}}]
+    [text/text opts (format-eth-value eth-value)]))
+
+(defn- label [size & text]
+  (let [opts {:weight :medium
+              :size (get-in sizes [:label size])
+              :style {:color colors/neutral-50}}]
+    [text/text opts (string/join \space text)]))
+
+(defn- divider [size]
+  (let [{:keys [width height]} (get-in sizes [:divider size])]
+    [rn/view {:style {:border-right-width width
+                      :border-right-color (colors/theme-colors colors/neutral-40 colors/neutral-50)
+                      :padding-left 8
+                      :align-self :center
+                      :height height}}]))
+
+(defn- rounded-container [size style & components]
+  (let [container-size (get-in sizes [:container size])
+        default-style {:accessibility-label :network-amount
+                       :background-color (colors/theme-colors colors/white colors/neutral-95)
+                       :border-radius (/ container-size 2)
+                       :padding 6}
+        opts {:style (merge default-style style)}]
+    (into [rn/view opts] components)))
+
+(defn- row [& components]
+  (into [rn/view {:style {:flex-direction :row
+                          :align-items :center}}]
+        components))
+
+(defn- icon-padding-top [size]
+  (if (= size :large) 1 3))
+
+(defn- icon-column [size icon]
+  (let [icon-size (get-in sizes [:icon size])]
+    [rn/view {:flex-direction :column
+              :align-self :flex-start
+              :padding-top (icon-padding-top size)
+              :padding-right 4}
+     [icon/icon icon {:size icon-size :color :none}]]))
+
+(defn- amount-column [size eth-value network-name]
+  [rn/view
+   [row
+    [amount-text size eth-value]
+    [divider size]]
+   [label size (i18n/label :t/on) network-name]])
+
+(defn view
+  "[network-amount/view opts]
+   opts
+   {:size         :small, :medium, :large ;; size of container, fonts and icon
+    :style        style                   ;; extra styles for the container (takes precedence)
+    :network-name \"Mainnet\"               ;; shown network name
+    :icon         :main-icons2/ethereum   ;; key of icon belonging to the network
+    :eth-value    1.2345678               ;; shown ETH value}"
+  [{:keys [size style network-name icon eth-value] :as _opts}]
+  [rounded-container size style
+   [row
```

Comment:
why do we need a separate `row` component ?

Reply:
and same question for `rounded-container`

Reply:
it just makes code more complex and confusing, better if its readable in one form here

Reply:
It's following the single responsibility principle. Although at first it may seem complex and confusing or unnecessary, in my experience, in the long run it's more maintainable and readable to extract functions, so that every function does one thing only. As the "social code" demonstrated in [Writing Quality Code in Erlang](https://youtu.be/CQyt9Vlkbis). I find it's more flexible, readable and maintainable that way. 

The downside is that to read the code for implementations you have to jump to the definitions (iff there is a need to know ). Advantages are that functions can be reused (e.g., `row` is used twice) and are encapsulated into clear names (e.g., the orchestrating component does not contain lower level implementation details like styling, but delegates this to lower level components like `rounded-container` and `row`, that could be extracted to a shared namespace.

If the preferred style is to create one component I am happy to change it. Also this component, especially when `size` is removed, is quite short.

I'll check what happens if I put everything (or maybe only the `rn/view`) top level.

Reply:
yes in theory it sounds good :)  but this component is so simple so its better to have all in one view

Reply:
Done.